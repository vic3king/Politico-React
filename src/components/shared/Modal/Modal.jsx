import React from 'react';
import PropTypes from 'prop-types';
import '../../../style/modal.scss';

const Modal = ({
  InputOne,
  InputTwo,
  Select,
  Header,
  SpanOne,
  SpanTwo,
  SpanThree,
  SpanFour,
  InputThree,
  ButtonOne,
  CloseModal,
}) => {
  return (
    <div>
      <section className="modal-main">
        <form className="modal-form">
          {Header}
          <div className="fix">
            {SpanOne}
            {Select}
          </div>
          <div className="fix">
            {SpanTwo}
            {InputOne}
          </div>
          <div className="fix">
            {SpanThree}
            {InputTwo}
          </div>
          <div className="fix">
            {SpanFour}
            {InputThree}
          </div>
          {ButtonOne}
          {CloseModal}
        </form>
      </section>
    </div>
  );
};

Modal.defaultProps = {
  InputOne: (PropTypes.defaultProps = ''),
  InputTwo: (PropTypes.defaultProps = ''),
  Select: (PropTypes.defaultProps = ''),
  Header: (PropTypes.defaultProps = ''),
  SpanOne: (PropTypes.defaultProps = ''),
  SpanTwo: (PropTypes.defaultProps = ''),
  SpanThree: (PropTypes.defaultProps = ''),
  ButtonOne: (PropTypes.defaultProps = ''),
  SpanFour: (PropTypes.defaultProps = ''),
  InputThree: (PropTypes.defaultProps = ''),
};
Modal.propTypes = {
  InputOne: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
  InputTwo: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
  Select: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
  Header: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
  SpanOne: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
  SpanTwo: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
  SpanThree: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
  ButtonOne: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
  InputThree: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
  SpanFour: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
};
export default Modal;
