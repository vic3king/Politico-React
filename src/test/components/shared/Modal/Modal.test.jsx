import React from 'react';
import { shallow } from 'enzyme';
import Modal from '../../../../components/shared/Modal/Modal';

describe('Modal component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Modal />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a form tag', () => {
    const wrapper = shallow(<Modal />);
    expect(wrapper.find('form'));
    expect(wrapper.hasClass('modal-main'));
  });
});
