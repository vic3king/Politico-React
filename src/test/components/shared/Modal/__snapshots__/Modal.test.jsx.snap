// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    ButtonOne=""
    Header=""
    InputOne=""
    InputThree=""
    InputTwo=""
    Select=""
    SpanFour=""
    SpanOne=""
    SpanThree=""
    SpanTwo=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="modal-main"
      >
        <form
          className="modal-form"
        >
          
          <div
            className="fix"
          >
            
            
          </div>
          <div
            className="fix"
          >
            
            
          </div>
          <div
            className="fix"
          >
            
            
          </div>
          <div
            className="fix"
          >
            
            
          </div>
          
        </form>
      </section>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="modal-form"
        >
          
          <div
            className="fix"
          >
            
            
          </div>
          <div
            className="fix"
          >
            
            
          </div>
          <div
            className="fix"
          >
            
            
          </div>
          <div
            className="fix"
          >
            
            
          </div>
          
        </form>,
        "className": "modal-main",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "",
            <div
              className="fix"
            >
              
              
            </div>,
            <div
              className="fix"
            >
              
              
            </div>,
            <div
              className="fix"
            >
              
              
            </div>,
            <div
              className="fix"
            >
              
              
            </div>,
            "",
            undefined,
          ],
          "className": "modal-form",
        },
        "ref": null,
        "rendered": Array [
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "",
                "",
              ],
              "className": "fix",
            },
            "ref": null,
            "rendered": Array [
              "",
              "",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "",
                "",
              ],
              "className": "fix",
            },
            "ref": null,
            "rendered": Array [
              "",
              "",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "",
                "",
              ],
              "className": "fix",
            },
            "ref": null,
            "rendered": Array [
              "",
              "",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "",
                "",
              ],
              "className": "fix",
            },
            "ref": null,
            "rendered": Array [
              "",
              "",
            ],
            "type": "div",
          },
          "",
          undefined,
        ],
        "type": "form",
      },
      "type": "section",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="modal-main"
        >
          <form
            className="modal-form"
          >
            
            <div
              className="fix"
            >
              
              
            </div>
            <div
              className="fix"
            >
              
              
            </div>
            <div
              className="fix"
            >
              
              
            </div>
            <div
              className="fix"
            >
              
              
            </div>
            
          </form>
        </section>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <form
            className="modal-form"
          >
            
            <div
              className="fix"
            >
              
              
            </div>
            <div
              className="fix"
            >
              
              
            </div>
            <div
              className="fix"
            >
              
              
            </div>
            <div
              className="fix"
            >
              
              
            </div>
            
          </form>,
          "className": "modal-main",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              <div
                className="fix"
              >
                
                
              </div>,
              <div
                className="fix"
              >
                
                
              </div>,
              <div
                className="fix"
              >
                
                
              </div>,
              <div
                className="fix"
              >
                
                
              </div>,
              "",
              undefined,
            ],
            "className": "modal-form",
          },
          "ref": null,
          "rendered": Array [
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "",
                  "",
                ],
                "className": "fix",
              },
              "ref": null,
              "rendered": Array [
                "",
                "",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "",
                  "",
                ],
                "className": "fix",
              },
              "ref": null,
              "rendered": Array [
                "",
                "",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "",
                  "",
                ],
                "className": "fix",
              },
              "ref": null,
              "rendered": Array [
                "",
                "",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "",
                  "",
                ],
                "className": "fix",
              },
              "ref": null,
              "rendered": Array [
                "",
                "",
              ],
              "type": "div",
            },
            "",
            undefined,
          ],
          "type": "form",
        },
        "type": "section",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
